src/main/java/io/camunda/testing/assertions/ProcessInstanceAssertions.java

```diff
@@ -388,4 +391,34 @@ public ProcessInstanceAssertions isNotWaitingForMessage(final String... messageN
             });
     return openMessageSubscriptions;
   }
+
+  public ProcessInstanceAssertions hasVariable(final String name) {
+    extractVariables()
+        .withFailMessage(
+            "Process with key %s does not contain variable with name %s",
+            actual.getProcessInstanceKey(), name)
+        .containsKey(name);
+    return this;
+  }
+
+  public ProcessInstanceAssertions hasVariableWithValue(final String name, final String value) {
+    extractVariables()
+        .withFailMessage(
+            "Process with key %s does not contain variable with name %s and value %s",
+            actual.getProcessInstanceKey(), name, value)
+        .containsEntry(name, value);
+    return this;
+  }
+
+  public MapAssert<String, String> extractVariables() {
+    final Map<String, String> variables =
+        StreamFilter.variable(recordStreamSource)
+            .withProcessInstanceKey(actual.getProcessInstanceKey())
+            .withRejectionType(RejectionType.NULL_VAL)
+            .stream()
+            .map(Record::getValue)
+            .collect(Collectors.toMap(VariableRecordValue::getName, VariableRecordValue::getValue));
```

Comment:
:x: Here we also need to unmarshall the JSON value to an object

Reply:
This may be a little difficult, because we don't know what object to expect. I wouldn't mind dropping this method for now and adding it after MVP if people request it.